### PR TITLE
Fix address already in use with webhooks input during reload 

### PR DESCRIPTION
**Pull request by DanKans**
Change HTTP listener by removing `ListenAndServe()` method and implementing `Close() net.Listen()` methods.
Change error reporting, by disabling adding it in case when `http.ErrServerClosed` occurred.

### Required for all PRs:

- [x] Signed [CLA](https://influxdata.com/community/cla/).
- [x] Associated README.md updated.
- [x] Has appropriate unit tests.

### CODE REVIEW

File: plugins/inputs/webhooks/webhooks.go [hunk 1]
```diff
@@ -105,11 +103,27 @@ func (wb *Webhooks) AvailableWebhooks() []Webhook {
...
+
+	for _, webhook := range wb.AvailableWebhooks() {
+		webhook.Register(r, acc)
+	}
+
+	wb.srv = &http.Server{Handler: r}
+
+	ln, err := net.Listen("tcp", fmt.Sprintf("%s", wb.ServiceAddress))
+	if err != nil {
+		acc.AddError(fmt.Errorf("E! Error starting server: %v", err))
```
Comment (danielnelson) [inline plugins/inputs/webhooks/webhooks.go]: I think we should return this error, making it fatal and preventing Telegraf from starting.  This probably indicates the port is in use or a privilege issue, and won't resolve, so I think exit is the right action even though it is technically a change in behavior.

File: plugins/inputs/webhooks/webhooks.go [hunk 2]
```diff
@@ -70,16 +73,11 @@ func (wb *Webhooks) Gather(_ telegraf.Accumulator) error {
...
-	r := mux.NewRouter()
-
-	for _, webhook := range wb.AvailableWebhooks() {
-		webhook.Register(r, acc)
-	}
-
-	err := http.ListenAndServe(fmt.Sprintf("%s", wb.ServiceAddress), r)
-	if err != nil {
-		acc.AddError(fmt.Errorf("E! Error starting server: %v", err))
+func (wb *Webhooks) listen(ln net.Listener, acc telegraf.Accumulator) {
```
Comment (danielnelson) [inline plugins/inputs/webhooks/webhooks.go]: Lets call this `serve` since we are already listening.

**Comment by danielnelson** [comment]
cc @francois2metz

File: plugins/inputs/webhooks/webhooks.go (hunk 2 above)
Comment (francois2metz) [reply plugins/inputs/webhooks/webhooks.go]: Or just inline it in the `Start` func.

**Comment by francois2metz** [comment]
@dankans thanks!

**Comment by DanKans** [comment]
Thanks @danielnelson and @francois2metz for help!

**Pull request merged**